specs/agents/configuration.md

```diff
@@ -143,3 +143,57 @@ By default, agents MUST send data to the APM Server at `http://localhost:8200/`.
 If possible, agents SHOULD detect sensible defaults for `service.name` and `service.version`.
 In any case agents MUST include `service.name` - if discovering it is not possible then the default value: `unknown-${service.agent.name}-service` MUST be used.
 This naming pattern allows the UI to display inline help on how to manually configure the name.
+
+### Adding a new configuration option
+
+#### Configuration option spec template
+
+When adding a new configuration option to the spec, please use the following template:
+
+```markdown
+### <option_name_in_central_config_notation> configuration
+
+The description of the option.
+Try to phrase it in a way that lets agents easily copy/paste the description into their configuration documentation page.
+This should also match the description in the [central configuration](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts) in the APM UI.
+
+|                |                                           |
+|----------------|-------------------------------------------|
+| Type           | See the Configuration Value Types section |
+| Default        | `<default value>`                         |
+| Central config | `<true/false>`                            |
+
+Additional description of the option that should not be part of the documentation.
+For example, whether agents SHOULD or MUST implement the option.
+
+```
+
+#### Adding to central configuration
+
+One of the things to determine when proposing to add a new configuration option is whether it should be added to central configuration.
+In general, we should try to make the majority of options available in central configuration.
+However, sometimes it might not be feasible if agents need to read the value at startup and changing the value at runtime is very difficult to achieve.
+
+When adding a centrally configurable option,
+the spec owner is expected to create a pull request in the Kibana repository to add the option to central configuration.
+As all the configuration options are added declaratively, this is no rocket science.
```

Comment:
```suggestion
As all the configuration options are added declaratively, this is not rocket science.
```

I'm also wondering if this sentence is necessary. Is the point to say that it's a straight-foward and simple task to add the option to the central configuration code in Kibana?

Reply:
Yes, that was the intent. I don't mind removing it, though.

Reply:
Let's just remove it, as it doesn't seem necessary to comment on how difficult it is to make changes to Kibana.